### PR TITLE
fix for double patters in webpack deploy

### PR DESCRIPTION
**Pull request by grzanka**

### CODE REVIEW

File: fixDeploy.mjs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import fse from 'fs-extra';
+import path from 'path';
+import { fileURLToPath, URL } from 'url';
+
+(async () => {
+	const repoRootUrl = new URL('./', import.meta.url);
+	const repoRoot = repoRootUrl.pathname;
+	const repoFolder = repoRoot.split('/').filter(Boolean).pop() ?? '/';
+	const sourceFolder = path.join(repoRoot, 'build/static/js');
+	// const destinationFolder = path.join(sourceFolder, 'static/js');
```
Comment (grzanka) [inline fixDeploy.mjs:12]:
```suggestion
```

File: fixDeploy.mjs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+	if (!fse.existsSync(sourceFolder)) {
+		throw new Error(
+			`Failed to handle webpack path bug. Source folder (${sourceFolder}) does not exist.`
+		);
+	}
+
+	// await fse.ensureDir(destinationFolder);
```
Comment (grzanka) [inline fixDeploy.mjs:20]:
```suggestion
```

File: fixDeploy.mjs [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+	const staticJsFiles = await fse.readdir(sourceFolder);
+
+	// Move all .js files that do not start with "main." or "web-worker." as those two get correctly loaded directly from static/js.
```
Comment (grzanka) [inline fixDeploy.mjs:23]:
```suggestion
```

are we now really moving the files somewhere ?

File: fixDeploy.mjs [hunk 4]
```diff
@@ -0,0 +1,36 @@
+// bug handler copied from https://github.com/facebook/create-react-app/issues/12503
```
Comment (grzanka) [inline fixDeploy.mjs:1]:
```suggestion
// bug handler copied from https://github.com/facebook/create-react-app/issues/12503
// needed to fix https://github.com/yaptide/ui/issues/640
```

File: fixDeploy.mjs [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+	// await fse.ensureDir(destinationFolder);
+	const staticJsFiles = await fse.readdir(sourceFolder);
+
+	// Move all .js files that do not start with "main." or "web-worker." as those two get correctly loaded directly from static/js.
+	const filesToFix = staticJsFiles.filter(val => /^(?!main\.|web-worker\.).+\.js$/.test(val));
+
+	for (const file of filesToFix) {
+		console.log(`Fixing ${file}`);
+		let code = await fse.readFile(path.join(sourceFolder, file), 'utf8');
+		code = code.split('static/js/').join('');
```
[the remark this answers is not in the document]
Comment (Derstilon) [reply fixDeploy.mjs:29]:
```suggestion
		// remove duplicate 'static/js' from paths 
		code = code.split('static/js/').join('');
		// replace yaptide_converter references with absolute paths 
```

File: fixDeploy.mjs (hunk 3 above)
Comment (Derstilon) [reply fixDeploy.mjs:23]:
```suggestion
	// Fix paths in all .js files that do not start with "main." or "web-worker." as those two get correctly loaded directly from static/js.
```

File: fixDeploy.mjs [hunk 6]
```diff
@@ -0,0 +1,37 @@
+// bug handler copied from https://github.com/facebook/create-react-app/issues/12503
+// needed to fix https://github.com/yaptide/ui/issues/640
+
+import fse from 'fs-extra';
+import path from 'path';
+import { fileURLToPath, URL } from 'url';
```
Comment (grzanka) [inline fixDeploy.mjs:6]:
```suggestion
import { URL } from 'url';
```

do we need `fileURLToPath` ?

File: fixDeploy.mjs [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+	const staticJsFiles = await fse.readdir(sourceFolder);
+
+	// Fix paths in all .js files that do not start with "main." or "web-worker." as those two get correctly loaded directly from static/js.
+	const filesToFix = staticJsFiles.filter(val => /^(?!main\.|web-worker\.).+\.js$/.test(val));
+
+	for (const file of filesToFix) {
+		console.log(`Fixing ${file}`);
+		let code = await fse.readFile(path.join(sourceFolder, file), 'utf8');
+		// remove duplicate 'static/js' from paths
```
Comment (grzanka) [inline fixDeploy.mjs:28]:
```suggestion
		// remove duplicate 'static/js' from paths 
		// in some chunk files the 'static/js' is already present in the path, therefore adding it by string concatenation results in corrupted paths with 'static/js/static/js' entries
		// here we solve that issues by replacing all occurrences of 'static/js' by empty string, so only single 'static/js' item will be propagated into chunks path
```

**Review by Derstilon — APPROVED** [review]

**Pull request merged**